tooling/metamodel-generator/src/main/java/org/hibernate/processor/ClassWriter.java

```diff
@@ -173,8 +173,7 @@ else if ( modifiers.contains( Modifier.PROTECTED ) ) {
 
 		final Element superTypeElement = entity.getSuperTypeElement();
 		if ( superTypeElement != null ) {
-			pw.print( " extends " +
-					entity.importType(getGeneratedSuperclassName( superTypeElement, entity.isJakartaDataStyle() )) );
+			pw.print( " extends " + getGeneratedSuperclassName( superTypeElement, entity.isJakartaDataStyle() ) );
```

Comment:
This should in principle not be necessary. `importType` is supposed to keep track of all the names it's already imported. 

I think what we instead need to do is "prime" the list of already-imported types with the FQN of the class we're generating, or something like that.

Reply:
Problem is that generated meta data class is not imported. My first version initially imported generated meta class to prevent class with identical simple name to be imported, but it seemed to me too complicated. Now I've reverted second version and applied first one. Check if that makes sense to you.

Reply:
>  Check if that makes sense to you.

Looks completely fine to me. What do you find complicated about it?

The reason I prefer this is that it avoids the use of a qualified name in the common case where the names *don't* collide.

Reply:
It is not so complicated (only one method that has been added), but my impression was that it is simpler to revert change I've made recently.